pkg/controller/performanceprofile/performanceprofile_controller.go

```diff
@@ -103,54 +208,225 @@ func (r *ReconcilePerformanceProfile) Reconcile(request reconcile.Request) (reco
-			Name:      cr.Name + "-pod",
-			Namespace: cr.Namespace,
-			Labels:    labels,
-		},
-		Spec: corev1.PodSpec{
-			Containers: []corev1.Container{
-				{
-					Name:    "busybox",
-					Image:   "busybox",
-					Command: []string{"sleep", "3600"},
-				},
-			},
-		},
+	if err := r.createOrUpdateMachineConfigPool(mcp); err != nil {
+		return err
+	}
+
+	// pause machine config pool
+	if err := r.pauseMachineConfigPool(mcp.Name, true); err != nil {
+		return err
+	}
+
+	// deploy machine config
+	mc, err := machineconfig.NewPerformance(r.assetsDir, profle)
+	if err != nil {
+		return err
+	}
+
+	if err := controllerutil.SetControllerReference(profle, mc, r.scheme); err != nil {
+		return err
+	}
+
+	if err := r.createOrUpdateMachineConfig(mc); err != nil {
+		return err
+	}
+
+	// deploy kubelet config
+	kc := kubeletconfig.NewPerformance(profle)
+	if err := controllerutil.SetControllerReference(profle, kc, r.scheme); err != nil {
+		return err
+	}
+	if err := r.createOrUpdateKubeletConfig(kc); err != nil {
+		return err
+	}
+
+	// deploy feature gate
+	fg := featuregate.NewLatencySensitive()
+	// TOOD: uncomment once https://bugzilla.redhat.com/show_bug.cgi?id=1788061 fixed
+	// if err := controllerutil.SetControllerReference(profle, fg, r.scheme); err != nil {
+	// 	return err
+	// }
+	if err := r.createOrUpdateFeatureGate(fg); err != nil {
+		return err
+	}
+
+	// deploy network latency tuned
+	networkLatencyTuned, err := tuned.NewNetworkLatency(r.assetsDir)
+	if err != nil {
+		return err
+	}
+
+	if err := controllerutil.SetControllerReference(profle, networkLatencyTuned, r.scheme); err != nil {
+		return err
+	}
+
+	if err := r.createOrUpdateTuned(networkLatencyTuned); err != nil {
+		return err
+	}
+
+	// deploy real time kernel tuned
+	realTimeKernelTuned, err := tuned.NewWorkerRealTimeKernel(r.assetsDir, profle)
+	if err != nil {
+		return err
+	}
+
+	if err := controllerutil.SetControllerReference(profle, realTimeKernelTuned, r.scheme); err != nil {
+		return err
+	}
+
+	if err := r.createOrUpdateTuned(realTimeKernelTuned); err != nil {
+		return err
+	}
+
+	// unpause machine config pool
+	return r.pauseMachineConfigPool(mcp.Name, false)
+}
+
+func (r *ReconcilePerformanceProfile) pauseMachineConfigPool(mcpName string, pause bool) error {
+	// TODO: in case if we will change nodeSelector to the machineConfigPoolSelector under the API,
+	// we will need to fetch the name of the machine config pool on runtime
+	updatedMcp, err := r.getMachineConfigPool(mcpName)
+	if errors.IsNotFound(err) {
+		klog.Warning("failed to set pause, the machine config pool does not exist, probably it was deleted")
+		return nil
+	}
+
+	if err != nil {
+		return err
+	}
+	if err != nil {
+		return err
+	}
+
+	updatedMcp.Spec.Paused = pause
+	return r.client.Update(context.TODO(), updatedMcp)
+}
+
+func (r *ReconcilePerformanceProfile) verifyPerformanceProfileParameters(performanceProfile *performancev1alpha1.PerformanceProfile) error {
```

Comment:
nit: not sure what the exact difference between the 2 words is, but mostly we talk about `validate` instead of `verify` (validation webhook) :thinking:

Reply:
changed